### PR TITLE
lsp-capf: convert from LSP snippet to yasnippet

### PR DESCRIPTION
**Pull request by kiennq**
This fix one of problem mentioned in #1451 
Specifically problem mentioned [here](https://github.com/emacs-lsp/lsp-mode/issues/1451#issuecomment-590695358)

We need to convert from LSP snippet format to yasnippet format when expanding it.

### CODE REVIEW

**Comment by s-kostyaev** [comment]
I've just checked it out. It works fine, thanks!

**Comment by yyoncho** [comment]
This breaks the CI, can you take a look?

File: lsp-mode.el [hunk 1]
```diff
@@ -4021,6 +4021,14 @@ PLIST is the additional data to attach to each candidate."
         (setq label-pos 0)))
     matches))
 
+(defun lsp--capf-to-yasnippet-snippet (text)
+  "Convert LSP snippet TEXT to yasnippet snippet."
+  ;; Need to escape "{"
+  (s-replace-regexp (rx (not (or "$" "\\")) (group "{"))
```
Comment (s-kostyaev) [inline lsp-mode.el]: I think here should be:
```elisp
(s-replace-regexp (rx (not (or "\$" "\\")) (group "{"))
```
It can fix CI for emacs < 27

**Comment by s-kostyaev** [comment]
Sorry, it was wrong tip. Correct one:
```elisp
(defun lsp--capf-to-yasnippet-snippet (text)
  "Convert LSP snippet TEXT to yasnippet snippet."
  ;; Need to escape "{"
  (s-replace-regexp (rx (not (any "$" "\\")) (group "{"))
                    (rx "\\" (backref 1))
                    text
                    nil nil 1))
```

thanks @larsbrinkhoff for https://hub.docker.com/r/larsbrinkhoff/emacsen/

**Pull request merged**